### PR TITLE
Create v2 api endpoint

### PR DESCRIPTION
**Pull request by cmgiven**
Connects #3708.

### CODE REVIEW

**Comment by cmgiven** [comment]
Note that this is basically just copying files from v1, just with a few attributes removed.

File: app/controllers/api/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class Api::V1::ApplicationController < ActionController::Base
+class Api::ApplicationController < ActionController::Base
```
Comment (amprokop) [inline app/controllers/api/application_controller.rb]: I'm guessing that you made a judgement call that this ApplicationController contains code that we expect not to change between versions at all, right? So we don't need to version it. Just checking.

Comment (cmgiven) [reply app/controllers/api/application_controller.rb]: Yes, that's right. I couldn't anticipate something getting added there that we would want to version.

**Comment by cmgiven** [comment]
I didn't see a need for feature flags, if that's what you're asking. I'm anticipating that we'll have Vets.gov hit the v2 endpoint through front-end development and into the future, and then once they've switched over, we can remove v1 from the codebase.

**Pull request merged**